infrastructure/balsa_queue/bq_main_macro.hh

```diff
@@ -12,39 +13,58 @@ void signal_handler(int s) {
-    sigaction(SIGINT, &sigIntHandler, NULL);                                  \
-    YAML::Node config;                                                        \
-    if (argc > 1) {                                                           \
-      try {                                                                   \
-        config = YAML::LoadFile(argv[1]);                                     \
-      } catch (YAML::BadFile e) {                                             \
-        std::cerr << "Could not find YAML file " << std::endl;                \
-        config = YAML::Node();                                                \
-      }                                                                       \
-    }                                                                         \
-    bq_type balsa_queue = bq_type();                                          \
-    if (config["bq_name"]) {                                                  \
-      balsa_queue.set_name(config["bq_name"].as<std::string>());              \
-      balsa_queue.gonogo().setName(config["bq_name"].as<std::string>());      \
-    } else {                                                                  \
-      balsa_queue.set_name(#bq_type);                                         \
-      balsa_queue.gonogo().setName(#bq_type);                                 \
-    }                                                                         \
-    balsa_queue.gonogo().nogo("init");                                        \
-    balsa_queue.set_comms_factory(std::make_unique<jet::MqttCommsFactory>()); \
-    balsa_queue.base_init();                                                  \
-    balsa_queue.init(config);                                                 \
-    while (!shutdown) {                                                       \
-      balsa_queue.base_loop();                                                \
-      balsa_queue.loop();                                                     \
-      usleep(balsa_queue.loop_delay_microseconds);                            \
-    }                                                                         \
-    balsa_queue.shutdown();                                                   \
-    return 0;                                                                 \
+void set_up_signal_handler() {
+  struct sigaction sigIntHandler;
+  sigIntHandler.sa_handler = signal_handler;
+  sigemptyset(&sigIntHandler.sa_mask);
+  sigIntHandler.sa_flags = 0;
+  sigaction(SIGINT, &sigIntHandler, NULL);
+}
+
+std::string parse_commandline_arguments(int argc, char* argv[]) {
+  if (argc > 1) {
+    return argv[1];
+  }
+  return "";
+}
+
+void parse_config(const std::string& config_path, Config& config) {
+  try {
+    config = YAML::LoadFile(config_path);
+  } catch (YAML::BadFile e) {
+    std::cerr << "Could not find YAML file. Using empty config." << std::endl;
```

Comment:
I've been burned a couple of times by trying to pass a real config, but accidentally passing an invalid path and then getting a failure at config loading time in the BQ.

Might make sense to always fail if the yaml at the passed path can't be opened

Reply:
Good idea. Switching to throwing a runtime exception if file not found.